### ISSUE TITLE
Store data to MongoDB

### ISSUE THREAD

**Issue by gabrielemilan** [simpleidserver/SimpleIdServer#27]
Is it possible to create a new project (like SimpleIdServer.Scim.Persistence.EF) to store data to MongoDB?

I created a project and I managed to store data to MongoDB, but now it doesn't work beacause of new feature in 1.1.1.

Do you have plan to support nosql database?

Thanks a lot.

**Comment by simpleidserver** [comment]
We will do some investigation, and implement a new library in the next release.

Kind regards

**Comment by simpleidserver** [comment]
A pre-release version of the library **SimpleIdServer.Scim.Persistence.MongoDB** is available in the feed : https://www.myget.org/F/advance-ict/api/v3/index.json.

Can-you please try again ?
You can find the sample project here : https://github.com/simpleidserver/SimpleIdServer/tree/master/src/Scim/SimpleIdServer.Scim.MongoDb.Startup

Kind regards

**Comment by gabrielemilan** [comment]
> A pre-release version of the library **SimpleIdServer.Scim.Persistence.MongoDB** is available in the feed : https://www.myget.org/F/advance-ict/api/v3/index.json.
> 
> Can-you please try again ?
> You can find the sample project here : https://github.com/simpleidserver/SimpleIdServer/tree/master/src/Scim/SimpleIdServer.Scim.MongoDb.Startup
> 
> Kind regards

Hello, yes i'm trying right now, I'll keep you posted. Good job :-) Thanks

**Comment by gabrielemilan** [comment]
Hello,

when I try to filter a user I get this error message:
![image](https://user-images.githubusercontent.com/10979448/79972119-b9c65d80-8495-11ea-9094-d3331621e0c7.png)

**Comment by simpleidserver** [comment]
Did-you fetch the latest changes ? 

This error occurred because the property "QueryableValuesInt" doesn't exist in the BSON representation. In the latest commit, we replaced the property "QueryableValuesInt" by "ValuesInteger" in the file **SCIMExpressionLinqExtensions.cs**, this exception should not occur.
You can find the changes here : https://github.com/simpleidserver/SimpleIdServer/commit/1820d844fbe5b83b86bc9482540af5c0dffe253e#diff-7c03340a18e0319b697da3d00c26da45

**Comment by gabrielemilan** [comment]
Sorry, I was using 1.1.1 version of library, let me check with the prerelease one.

**Comment by gabrielemilan** [comment]
Ok, it seems working properly 👍 

I noticed that the api i svery slow, I debugged a bit and it seems that the problem is the new instances of MongoClient in each repository.

My suggestion is to create a singleton like:

```
services.AddSingleton<IMongoClient>(s =>new MongoClient(options.ConnectionString));
services.AddSingleton(provider=>provider.GetService<IMongoClient).GetDatabase(options.Database));
```

And inject the database in repository class:
```
private readonly IMongoDatabase _database;
        public SCIMRepresentationQueryRepository(IMongoDatabase database)
        {
            _database = database;
        }
```

**Comment by simpleidserver** [comment]
Thank you for the suggestions.

We didn't take the time to finalize the library, don't be surprise if there are some bugs.
We will do it ASAP (this week) and publish the library in the next release.

**Comment by gabrielemilan** [comment]
Ok cool, another thing. Is it possible to make customizable the name of collections?

**Comment by simpleidserver** [comment]
Ok we will take into consideration your remark for the next release :)

**Comment by gabrielemilan** [comment]
I got another error:
![image](https://user-images.githubusercontent.com/10979448/80080882-81388980-8552-11ea-8c6a-d126637d7ef5.png)

when I try to patch a group inserting new member:

```
curl --location --request PATCH 'https://localhost:5001/Groups/f7333af0-3c36-4823-b5a0-f31a718fccbc' \
--header 'Content-Type: application/json' \
--data-raw '{
    "schemas": [
        "urn:ietf:params:scim:api:messages:2.0:PatchOp"
    ],
    "Operations": [
        {
            "op": "add",
            "path": "members",
            "value": [
                {
                    "display": "User 1",
                    "value": "4f56444d-9022-4319-b6d6-283afed3f08d"
                }
            ]
        }
    ]
}'
```

**Comment by simpleidserver** [comment]
The issue should be fixed in the branch "release/1.1.2", can-you please try again ?
